tls/s2n_connection.c

```diff
@@ -78,6 +82,11 @@ struct s2n_connection *s2n_connection_new(s2n_mode mode)
     conn->managed_io = 0;
     conn->corked_io = 0;
 
+    /* By default, only the client will authenticate the Server's Certificate. The Server does not request or
+     * authenticate any client certificates. */
+    conn->client_cert_auth_type = S2N_CERT_AUTH_NONE;
+    conn->verify_cert_chain_callback = deny_all_certs;
```

Comment:
Why not set a default callback that matches the previous behavior of client mode (accept_all_certs or something similar)? If I'm following this PR correctly, all previous client code will break(s2nc included). Having a `accept_all_certs` default would allow us to avoid defining/setting `accept_all_rsa_certs` for the self talk unit tests.